File: scalardb-sql-jdbc-sample/src/main/java/sample/command/GetOrderCommand.java
package sample.command;

import java.util.concurrent.Callable;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;
import sample.Sample;

@Command(name = "GetOrder", description = "Get order information by order ID")
public class GetOrderCommand implements Callable<Integer> {

  @Parameters(index = "0", paramLabel = "ORDER_ID", description = "order ID")
  private String orderId;

  @Override
  public Integer call() throws Exception {
    System.out.println(new Sample().getOrderByOrderId(orderId));
    return 0;
  }
}

File: scalardb-dotnet-samples/scalardb-cluster-sql-sample/Commands/GetOrderCommand.cs
using System.CommandLine;

namespace ScalarDbClusterSqlSample.Commands;

public static class GetOrderCommand
{
    private const string Name = "GetOrder";
    private const string Description = "Get order information by order ID";

    private const string ArgName = "id";
    private const string ArgDescription = "order ID";

    public static Command Create()
    {
        var orderIdArg = new Argument<string>(ArgName, ArgDescription);
        var getOrderCommand = new Command(Name, Description)
        {
            orderIdArg
        };

        getOrderCommand.SetHandler(async orderId =>
        {
            using var sample = new Sample();
            var order = await sample.GetOrderByOrderId(orderId);
            
            Console.WriteLine(order);
        }, orderIdArg);
    
        return getOrderCommand;
    }
}
